dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/analytics/dimension/mappers/ProgramStageDataElementMapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.dimension.mappers;
+
+import java.util.Collection;
+
+import lombok.Getter;
+
+import org.hisp.dhis.analytics.dimension.DimensionResponse;
+import org.hisp.dhis.common.BaseIdentifiableObject;
+import org.hisp.dhis.hibernate.HibernateProxyUtils;
+import org.hisp.dhis.program.ProgramStageDataElement;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.ImmutableSet;
+
+@Service
+public class ProgramStageDataElementMapper extends BaseDimensionalItemObjectMapper
+{
+
+    @Getter
+    private final Collection<Class<? extends BaseIdentifiableObject>> supportedClasses = ImmutableSet.of(
+        ProgramStageDataElement.class );
+
+    @Override
+    public DimensionResponse map( BaseIdentifiableObject dimension )
+    {
+        ProgramStageDataElement programStageDataElement = (ProgramStageDataElement) dimension;
+
+        return super.map( HibernateProxyUtils.unproxy( programStageDataElement.getDataElement() ) )
```

Comment:
The use of `HibernateProxyUtils` is a bit suspicious as we are exposing the ORM in services. Is there any way to refactor this code to avoid it?

Reply:
you are right, this is not even necessary